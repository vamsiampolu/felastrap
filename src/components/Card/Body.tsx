import { useFela } from "react-fela";
import type { DivProps, StyleProps } from "../common";
import { overlayBase, overlayLayoutRule } from "./imageStyle";

const padding = {
  paddingTop: "1rem",
  paddingBottom: "1rem",
  paddingLeft: "1rem",
  paddingRight: "1rem",
};

const body = ({ color }: StyleProps<CardBodyProps>) => ({
  ...padding,
  color,
  flexGrow: 1,
  flexShrink: 1,
  flexBasis: "auto",
});

export type CardBodyProps = DivProps & { color?: string; overlay?: boolean };

export const CardBody = ({
  className = "",
  children,
  color = "#000",
  overlay = false,
  ...other
}: CardBodyProps) => {
  const { css } = useFela({ color, overlay });
  const cx = `${css([body, overlayBase, overlayLayoutRule])} ${
    className || ""
  }`.trim();
  return (
    <div data-testid="card-body" className={cx} {...other}>
      {children}
    </div>
  );
};
