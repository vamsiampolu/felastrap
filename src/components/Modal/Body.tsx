import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  position: "relative",
  padding: "15px",
  borderBottom: "1px solid #dee2e6",
});

const scrollableRule = ({ scrollable = false }) =>
  scrollable
    ? {
        overflowY: "auto !important",
        minHeight: "1500px",
      }
    : {};

export const ModalBody = ({
  className = "",
  children,
  scrollable = false,
  ...props
}: DivProps & { scrollable?: boolean }) => {
  const { css } = useFela({ scrollable });
  const cx = `${css([rule, scrollableRule])} bs-modal-body ${className}`.trim();
  return (
    <div data-testid="modal-body" className={cx} {...props}>
      {children}
    </div>
  );
};
