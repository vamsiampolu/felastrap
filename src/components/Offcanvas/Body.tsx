import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  flexGrow: 1,
  padding: "1rem 1rem",
  overflowY: "auto",
});

export const OffcanvasBody = ({
  className = "",
  children,
  ...props
}: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} offcanvas-body ${className}`.trim();
  return (
    <div data-testid="offcanvas-body" className={cx} {...props}>
      {children}
    </div>
  );
};
