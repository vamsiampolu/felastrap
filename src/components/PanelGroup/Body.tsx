import { useFela } from "react-fela";
import { DivProps } from "../common";
import { Content } from "../Collapse/Content";

const flushBody = ({ flush = false }) => {
  return flush ? { borderWidth: 0 } : {};
};

const rule = () => ({
  padding: "1rem 1.25rem",
});

export type BodyProps = DivProps & { flush?: boolean };
export const Body = ({
  className = "",
  flush = false,
  children,
  ...other
}: BodyProps) => {
  const { css } = useFela({ flush });
  const cx = `${css([rule, flushBody])} panel-group-body ${className}`;
  return (
    <div data-testid="panel-group-body" className={cx} {...other}>
      {children}
    </div>
  );
};

export const BodyContainer = ({ className = "", id, ...props }: DivProps) => {
  return (
    <Content id={id}>
      <Body className={className} {...props} />
    </Content>
  );
};
