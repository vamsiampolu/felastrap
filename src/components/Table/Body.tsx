import { IStyle } from "fela";
import { ComponentPropsWithoutRef, useContext } from "react";
import { useFela } from "react-fela";
import { TableContext } from "./TableContext";
import { hoveredBackgroundColorByVariant } from "./variant";
import { Variant } from "../common";

export type TableBodyProps = ComponentPropsWithoutRef<"tbody"> & {
  striped?: boolean;
  hover?: boolean;
  bordered?: boolean;
};

const rule = () => ({ verticalAlign: "inherit" });

const hoverRule = ({ hover = false, variant = "default" }): IStyle =>
  hover
    ? {
        "> tr:hover > *": {
          cursor: "pointer",
          color: "#000",
          backgroundColor: hoveredBackgroundColorByVariant[variant as Variant],
        },
      }
    : {};

const borderedRule = ({ bordered = false }) =>
  bordered
    ? {
        borderWidth: "1px 0",
      }
    : {};

export const TableBody = ({
  className = "",
  children,
  ...props
}: TableBodyProps) => {
  const { hover, bordered, variant = "default" } = useContext(TableContext);
  const { css } = useFela({ hover, bordered, variant });
  const cx = `${css([
    rule,
    hoverRule,
    borderedRule,
  ])} table-body ${className}`.trim();
  return (
    <tbody data-testid="table-body" className={cx} {...props}>
      {children}
    </tbody>
  );
};
