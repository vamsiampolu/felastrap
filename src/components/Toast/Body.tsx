import { useFela } from "react-fela";
import { DivProps } from "../common";

const rule = () => ({
  padding: "0.75rem",
  wordWrap: "break-word",
});

export const ToastBody = ({ children, className = "", ...props }: DivProps) => {
  const { css } = useFela();
  const cx = `${css(rule)} toast-body ${className}`.trim();
  return (
    <div data-testid="toast-body" className={cx} {...props}>
      {children}
    </div>
  );
};
