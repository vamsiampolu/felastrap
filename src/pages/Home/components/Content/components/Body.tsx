import { ComponentPropsWithoutRef, ReactNode } from "react";
import { useFela } from "react-fela";
import { IconLink } from "./IconLink";
import { ArrowRightIcon } from "../../../../../components/Icons/ArrowRightIcon";

const articleBody = () => ({
  display: "flex",
  fontSize: "1rem",
  fontWeight: 400,
  "@media((min-width: 768px)": {
    fontSize: "1.25rem",
  },
  marginBottom: 0,
  justifyContent: "flex-start",
  "@media (min-width: 768px)": {
    fontSize: "1.25rem",
  },
});

export const BodyParagraph = ({
  className = "",
  children,
  linkText,
  href,
  ...props
}: ComponentPropsWithoutRef<"p"> & { href: string; linkText: ReactNode }) => {
  const { css } = useFela();
  const cx = `${css(articleBody)} column-body ${className}`.trim();
  return (
    <p className={cx} {...props}>
      <IconLink href={href}>
        {linkText}
        <ArrowRightIcon />
      </IconLink>
      {children}
    </p>
  );
};
