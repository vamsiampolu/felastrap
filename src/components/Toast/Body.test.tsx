import { IRenderer } from "fela";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../test-helpers";
import { ToastBody } from "./Body";
import { render, screen } from "@testing-library/react";

describe("components/Toast/Body", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a ToastBody", async () => {
    const component = <ToastBody>Toast Body</ToastBody>;
    const wrappedComponent = <Wrapper renderer={renderer}>{component}</Wrapper>;

    render(wrappedComponent);
    expect(screen.getByTestId("toast-body")).toBeInTheDocument();

    expect(screen.queryByTestId("toast-body")).toHaveTextContent("Toast Body");
    expect(screen.queryByTestId("toast-body")).toHaveClass("toast-body");

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a ToastBody with a className", async () => {
    const component = (
      <ToastBody className="chatbox-notification">Toast Body</ToastBody>
    );
    const wrappedComponent = <Wrapper renderer={renderer}>{component}</Wrapper>;

    render(wrappedComponent);
    expect(screen.getByTestId("toast-body")).toBeInTheDocument();

    expect(screen.queryByTestId("toast-body")).toHaveTextContent("Toast Body");
    expect(screen.queryByTestId("toast-body")).toHaveClass(
      "toast-body chatbox-notification",
    );

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
