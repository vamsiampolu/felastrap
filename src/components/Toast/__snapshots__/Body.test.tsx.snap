// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Toast/Body should render a ToastBody 1`] = `
".a {
  word-wrap: break-word;
}
.b {
  padding-top: 0.75rem;
}
.c {
  padding-right: 0.75rem;
}
.d {
  padding-bottom: 0.75rem;
}
.e {
  padding-left: 0.75rem;
}


<div data-testid="toast-body" class="a b c d e toast-body">Toast Body</div>
;
"
`;

exports[`components/Toast/Body should render a ToastBody with a className 1`] = `
".a {
  word-wrap: break-word;
}
.b {
  padding-top: 0.75rem;
}
.c {
  padding-right: 0.75rem;
}
.d {
  padding-bottom: 0.75rem;
}
.e {
  padding-left: 0.75rem;
}


<div data-testid="toast-body" class="a b c d e toast-body chatbox-notification">
  Toast Body
</div>
;
"
`;
